.github/workflows/validate-migration.yaml

```diff
@@ -73,29 +171,43 @@ jobs:
           mkdir -p vendor/helix-core-api/linux
           tar -C vendor/helix-core-api/linux -xzf p4.tgz --strip 1
 
-      - name: "Cache OpenSSL"
+      - name: "Cache OpenSSL source"
+        id: cache-openssl-source
         uses: actions/cache@v2
-        id: cache-openssl
         with:
-          path: ${{ env.OPENSSL_INSTALL_DIR }}
-          key: ${{ runner.os }}-openssl-${{ env.OPENSSL_DOWNLOAD_URL }}
+          path: ${{ env.OPENSSL_SOURCE_DIR }}
+          key: ${{ runner.os }}-openssl-source-${{ env.OPENSSL_DOWNLOAD_URL }}
 
-      - name: "Compile OpenSSL"
-        if: ${{ steps.cache-openssl.outputs.cache-hit != 'true' }}
+      - name: "Download OpenSSL source"
+        if : ${{ steps.cache-openssl-source.outputs.cache-hit != 'true' }}
         run: |
-          mkdir openssl-src
           curl -L "${OPENSSL_DOWNLOAD_URL}" --output openssl.tgz
-          tar -C openssl-src -xzf openssl.tgz --strip 1
-          pushd openssl-src
-          ./config --prefix="${OPENSSL_INSTALL_DIR}" --static
-          make install
-          popd 
+          mkdir -p "${OPENSSL_SOURCE_DIR}" || true
+          tar -C "${OPENSSL_SOURCE_DIR}" -xzf openssl.tgz --strip-components 1
+          rm openssl.tgz || true
+
+      - name: "Compile OpenSSL"
+        run: |
+          .github/workflows/validate-migration-scripts/compile-openssl.sh
+        env:
+          CC: "sccache clang"
+          CXX: "sccache clang++"
 
       - name: "Run validate migration script"
         run: |
-          sudo apt-get install gettext --yes # for envsubst 
-          .github/workflows/run-test.sh
+          .github/workflows/validate-migration-scripts/run-test.sh
         env:
-          P4CLIENT: "validate-migration-${{ github.ref_name }}-${{ github.run_number }}-${{ matrix.sanitizer }}" # the temporary Perforce client name to use while validating the git migration
+          P4CLIENT: "validate-migration-${{ github.ref_name }}-${{ github.sha }}-${{ github.run_number }}-${{ matrix.tool }}" # the temporary Perforce client name to use while validating the git migration
           OPENSSL_ROOT_DIR: ${{ env.OPENSSL_INSTALL_DIR }}
-          CMAKE_CXX_FLAGS: "${{ matrix.CMAKE_CXX_FLAGS }} -O0"
\ No newline at end of file
+          CMAKE_CXX_COMPILER: "clang++"
+          CMAKE_C_COMPILER: "clang"
+          CMAKE_C_COMPILER_LAUNCHER: "sccache"
+          CMAKE_CXX_COMPILER_LAUNCHER: "sccache"
+          CMAKE_CXX_FLAGS: "${{ env.CMAKE_CXX_FLAGS }} -stdlib=libc++ -fno-omit-frame-pointer -O0"
```

Comment:
Suggestion: Please consolidate all the arguments being set within a single step/job. E.g. where I suggested adding `-O1` earlier, you can add the `-stdlib` and `-fno-omit-frame-pointer` args there.